pennylane_lightning/src/algorithms/JacobianProd.hpp

```diff
@@ -117,7 +117,7 @@ template <class T = double> class VectorJacobianProduct {
                     -> std::vector<T> { return std::vector<T>(num_params, 0); };
         }
 
-        return [=](const JacobianData<T> &jd) -> std::vector<T> {
+        return [=, this](const JacobianData<T> &jd) -> std::vector<T> {
```

Comment:
Does the `this` capture change in C++20?

Reply:
Yes, C++20 deprecated the implicit capture of *this.

Reply:
https://www.nextptr.com/tutorial/ta1430524603/capture-this-in-lambda-expression-timeline-of-change says "The value-capture-default can implicitly capture this or &(*this). However, the implicit capture of this through [=] is deprecated in C++20."